std-bits/table/src/main/java/org/enso/table/data/index/UnorderedMultiValueKey.java

```diff
@@ -0,0 +1,81 @@
+package org.enso.table.data.index;
+
+import java.util.Objects;
+import org.enso.table.data.column.storage.Storage;
+import org.enso.table.text.TextFoldingStrategy;
+import org.enso.table.text.UnicodeNormalizedFold;
+
+/**
+ * A multi-value key for unordered operations like group-by or distinct.
+ *
+ * <p>It relies on folding logic that coerces values to their representatives in such a way that
+ * their equality is consistent with how Enso would handle equality.
+ *
+ * <p>As it relies on hashing, it currently is not prepared to work correctly for custom
+ * Enso-defined objects, as hashing of such objects is not yet implemented properly.
+ */
+public class UnorderedMultiValueKey extends MultiValueKeyBase {
```

Comment:
We should probably throw an exception is an unsupported object is put into this key.

Reply:
Ok.